### PR TITLE
[browserslist, autoprefixer] Remove types

### PR DESCRIPTION
**Pull request by Semigradsky**
Please fill in this template.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/autoprefixer/package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
     "private": true,
     "dependencies": {
-        "postcss": "7.x.x"
+        "browserslist": "^4.15.x",
```
Comment (peterblazejewicz) [inline types/autoprefixer/package.json:4]:
```suggestion
        "browserslist": "^4.15",
```
Also note, you have to open PR to MS tools, to allow install this via `package.json` dependency:
https://github.com/DefinitelyTyped/DefinitelyTyped#removing-a-package

File: types/autoprefixer/package.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
+        "postcss": "^8.x.x"
```
Comment (peterblazejewicz) [inline types/autoprefixer/package.json:5]:
```suggestion
        "postcss": "^8"
```

**Comment by peterblazejewicz** [comment]
x-ref:
https://github.com/browserslist/browserslist/pull/544

**Review by peterblazejewicz — COMMENTED** [review]
@Semigradsky thx, looking good, but you need to open that toolling PR
and it needs to referrence this PR, your CI once reset after white source 
addition, will succeed and no other steps are required

**Comment by peterblazejewicz** [comment]
BTW: you can make this published, not-draft, looks OK

**Comment by Semigradsky** [comment]
@peterblazejewicz thanks for your great advices! I think about moving `autoprefixer` typings to https://github.com/postcss/autoprefixer so I will remain this as draft

**Review by peterblazejewicz — APPROVED** [review]

**Pull request merged**